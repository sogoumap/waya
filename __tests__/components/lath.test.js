import 'babel-polyfill';
import { shallowMount,createLocalVue,mount } from '@vue/test-utils';
import { render } from '@vue/server-test-utils';
import Lath from '../../src/components/lath';

const LocalVue = createLocalVue();

describe('Logic Of Lath', () => {
  const Wrapper = shallowMount(Lath, {
    localVue: LocalVue
  });

  test('border', () => {
    expect(Wrapper.contains('.wayo-lath__separator-top')).toBe(false);
    expect(Wrapper.contains('.wayo-lath__separator-bottom')).toBe(false);

    Wrapper.setProps({
      borderTop: true,
      borderBottom: true
    });
    expect(Wrapper.contains('.wayo-lath__separator-top')).toBe(true);
    expect(Wrapper.contains('.wayo-lath__separator-bottom')).toBe(true);
  });

  test('inline', () => {
    expect(Wrapper.classes()).not.toContain('wayo-lath_inline');
    expect(Wrapper.classes()).not.toContain('wayo-lath_icon');
    expect(Wrapper.classes()).not.toContain('wayo-lath_content_header');
    Wrapper.setProps({
      inline: true
    });
    expect(Wrapper.classes()).toContain('wayo-lath_inline');
  });

  test('icon', () => {
    expect(Wrapper.contains('.wayo-lath__icon-head')).toBe(false);

    Wrapper.setProps({
      icon: 'bus',
      iconPosition: 'top'
    });
    expect(Wrapper.contains('.wayo-lath__icon-head')).toBe(true);
    expect(Wrapper.contains('.wayo-lath__icon-head_top')).toBe(true);
  });

  test('styles', () => {
    expect(Wrapper.vm.styles).toBe('');
    
    Wrapper.setData({
      tail: {
        icon: 'bus'
      }
    });
    expect(Wrapper.vm.styles).toMatch(/padding\-right:35px;/);

    Wrapper.setData({
      tail: {
        icon: 'bus',
        label: 'tail'
      }
    });
    expect(Wrapper.vm.styles).toMatch(/padding\-right:95px;/);

    Wrapper.setData({
      hasHeader: true,
      hasContent: false
    });
    expect(Wrapper.classes()).not.toContain('wayo-lath_content_header');

    Wrapper.setData({
      hasHeader: true,
      hasContent: true
    });
    expect(Wrapper.classes()).toContain('wayo-lath_content_header');
  });
});

describe('Snapshot of Lath', () => {
  const Wrapper = mount(Lath, {
    localVue: LocalVue
  });

  test('',() => {
    expect(Wrapper).toMatchSnapshot();
  });
});

describe('Logic Of LathHeader', () => {
  const Wrapper = shallowMount(Lath.Header, {
    localVue: LocalVue
  });

  test('display',()=>{
    expect(Wrapper.vm.show).toBe(false);

    Wrapper.setProps({
      title: 'title',
      subtitle: 'subtitle'
    });
    expect(Wrapper.vm.show).toBe(true);

    Wrapper.setProps({
      title: 'title',
      subtitle: ''
    });
    expect(Wrapper.vm.show).toBe(true);

    Wrapper.setProps({
      title: '',
      subtitle: 'subtitle'
    });
    expect(Wrapper.vm.show).toBe(true);
  });
});

describe('Snapshot of LathHeader', () => {
  const Wrapper = mount(Lath.Header, {
    localVue: LocalVue,
    propsData: {
      title: 'title',
      subtitle: 'subtitle'
    }
  });

  test('',() => {
    expect(Wrapper).toMatchSnapshot();
  });
});

describe('Logic Of LathContent', () => {
  const Wrapper = shallowMount(Lath.Content, {
    localVue: LocalVue,
    propsData: {
      contentLineLimit: 2
    }
  });

  test('attributes',()=>{
    expect(Wrapper.vm.contentMaxHeight).toBe(42);
    expect(Wrapper.vm.show).toBe(false);

    Wrapper.setProps({
      content: 'content'
    });
    expect(Wrapper.vm.show).toBe(true);
    expect(Wrapper.vm.single_line_content).toBe(false);
    expect(Wrapper.vm.content_oversize).toBe(false);
  });

  test('methods',()=>{
    Wrapper.setData({
      content_oversize: true
    });
    Wrapper.vm.expandContent();
    expect(Wrapper.vm.content_oversize).toBe(false);
  });
});

describe('Snapshot of LathContent', () => {
  const Wrapper = mount(Lath.Content, {
    localVue: LocalVue,
    propsData: {
      content: 'content'
    }
  });

  test('',() => {
    expect(Wrapper).toMatchSnapshot();
  });
});

describe('Logic Of LathTail', () => {
  const Wrapper = shallowMount(Lath.Tail, {
    localVue: LocalVue
  });

  test('attributes',()=>{
    expect(Wrapper.vm.show).toBe(false);

    Wrapper.setProps({
      icon: 'bus'
    });
    expect(Wrapper.vm.show).toBe(true);
  });
});

describe('Snapshot of LathTail', () => {
  const Wrapper = render(Lath.Tail, {
    localVue: LocalVue,
    propsData: {
      icon: 'bus',
      label: 'tail'
    }
  });

  test('',() => {
    expect(Wrapper).toMatchSnapshot();
  });
});