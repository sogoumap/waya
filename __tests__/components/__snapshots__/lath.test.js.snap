// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Lath  1`] = `<div class="wayo-lath"></div>`;

exports[`Snapshot of LathContent  1`] = `
<div class="wayo-lath__content">
  <p class="wayo-lath__content-entity wayo-lath__content_single-line">
    content
  </p>
  <!---->
  <!---->
</div>
`;

exports[`Snapshot of LathHeader  1`] = `<div class="wayo-lath__head"><span class="wayo-lath__title">title</span> <span class="wayo-lath__subtitle" style="margin-top: 5px;">subtitle</span></div>`;

exports[`Snapshot of LathTail  1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-lath__tail",
      "data-server-rendered": "true",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "wayo-lath__tail-label",
        },
        "children": Array [
          Object {
            "data": "tail",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": " ",
          "next": Object {
            "attribs": Object {
              "class": "wayo-lath__tail-icon",
              "color": "",
              "name": "bus",
            },
            "children": Array [],
            "name": "wayo-icon",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "color": undefined,
              "name": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "color": undefined,
              "name": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "data": " ",
        "next": Object {
          "attribs": Object {
            "class": "wayo-lath__tail-icon",
            "color": "",
            "name": "bus",
          },
          "children": Array [],
          "name": "wayo-icon",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "color": undefined,
            "name": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "color": undefined,
            "name": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "wayo-lath__tail-label",
          },
          "children": Array [
            Object {
              "data": "tail",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "text",
      },
      Object {
        "attribs": Object {
          "class": "wayo-lath__tail-icon",
          "color": "",
          "name": "bus",
        },
        "children": Array [],
        "name": "wayo-icon",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": " ",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "wayo-lath__tail-label",
            },
            "children": Array [
              Object {
                "data": "tail",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "color": undefined,
          "name": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "color": undefined,
          "name": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
